File: app/src/main/java/com/my/test2/MainActivity.java
package com.my.test2;

import com.my.test2.databinding.ActivityMainBinding;

public class MainActivity extends BaseActivity<ActivityMainBinding,MainModel> {

    @Override
    protected void onPrepare() {

    }

    @Override
    protected ActivityMainBinding getBinding() {
        return ActivityMainBinding.inflate(getLayoutInflater());
    }

    @Override
    protected MainModel getModel() {
        return new MainModel(binding,this);
    }
}

File: app/src/main/java/com/my/test2/MainActivity.kt
package com.my.test2

import com.my.test2.databinding.ActivityMainBinding
import com.my.test2.kotlinbase.BaseActivity

class MainActivity: BaseActivity<ActivityMainBinding, MainModel>() {
    override fun onPrepare() {

    }

    override fun getBindingInstance(): ActivityMainBinding {
       return ActivityMainBinding.inflate(layoutInflater)
    }

    override fun getModelInstance(): MainModel {
        return MainModel(binding,this)
    }
}